grails-app/migrations/views/pending_cycle_count_request.sql

```diff
@@ -15,24 +15,23 @@ CREATE OR REPLACE VIEW pending_cycle_count_request AS
         product_classification.abc_class as abc_class,
         SUM(pa.quantity_on_hand) AS quantity_on_hand,
         SUM(CASE WHEN pa.quantity_on_hand < 0 THEN 1 ELSE 0 END) as negative_item_count,
+        -- A comma-separated list of internal locations in the format <zone>:<bin location name>
         GROUP_CONCAT(DISTINCT (
-            CASE
-                WHEN pa.quantity_on_hand != 0 THEN pa.bin_location_name
-                ELSE NULL
-                END
-            )
-        ) as internal_locations
+            IF(pa.quantity_on_hand != 0, CONCAT_WS(':', zone.name, pa.bin_location_name), NULL)
+        )) as internal_locations
     FROM
         cycle_count_request ccr
             LEFT OUTER JOIN product_availability pa
                 ON ccr.product_id = pa.product_id
                 AND ccr.facility_id = pa.location_id
             LEFT OUTER JOIN cycle_count cc
                 ON ccr.cycle_count_id = cc.id
-            JOIN location ON pa.location_id = location.id
+            JOIN location facility ON pa.location_id = facility.id
+            LEFT OUTER JOIN location bin_location ON pa.bin_location_id = bin_location.id
+            LEFT OUTER JOIN location zone ON bin_location.zone_id = zone.id
```

Comment:
Have you checked the performance with/without your changes? There are a lot of new joins, so I am curious how it looks right now :thinking:

Reply:
it's hard for me to test on my local since the performace is all over the place. It doesn't seem to be significantly slower than it was before but it was already really quite slow so I hope the task for materializing the views will help with that